src/transformers/models/gpt2/modeling_gpt2.py

```diff
@@ -717,18 +717,22 @@ def forward(
                 all_hidden_states = all_hidden_states + (hidden_states,)
 
             if getattr(self.config, "gradient_checkpointing", False):
+                if use_cache:
```

Comment:
I agree with this change, I don't see how and why `use_cache` could be used with `gradient_checkpointing`. The only problem I can see if someone trains GPT2 with gradient checkpointing, but wants to enable `use_cache` for validation and evaluation afterward. IMO, we should however force `use_cache=False` during train, and `gradient_checkpointing=False` during validation  in the `Trainer`? Let's see what @LysandreJik and @sgugger think

Reply:
Why would `gradient_checkpointing` be `True` during validation? I'm a bit lost.

Reply:
Explained offline